spec/requests/user_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.describe 'User Management', type: :request do
+  let(:signup) do
+    get '/signup'
+    expect(response).to render_template(:new)
+    post '/users', params: { user: { name: @name } }
+  end
+  let(:event) { Event.create(title: 'This is a valid title') }
+
+  context 'when user signs up with valid name' do
+    it 'logs in user and loads root' do
+      @name = 'Lucas'
+      signup
+      user = User.find_by(name: 'Lucas')
+      expect(response).to redirect_to('/events')
+      expect(session[:user_id]).to eql(user.id)
+    end
```

Comment:
Sorry but the integration tests, to my knowledge, doesn't deal with model.
The good to do it, it's to check a part of the web page resulting from Sign Up and execpt for the user name somewhere. Also, you can show that the **Logout** which were not present, is present now.